File: JAVA/JAVA_DSA_QS_EASY/PROB_3033_Modify_The_Matrix.java
class Solution {
    public int[][] modifiedMatrix(int[][] matrix) {
        int n = matrix[0].length;
        int m = matrix.length;
        int[][] fx = new int[n][m];
        int[][] rx = new int[m][n];
        for (int j = 0; j < n; j++) {
            for (int k = 0; k < m; k++) {
                fx[j][k] = matrix[k][j];
            }
        }
        for (int i = 0; i < fx.length; i++) {
            int max = Integer.MIN_VALUE;
            for (int value : fx[i]) {
                if (value != -1) {
                    max = Math.max(max, value);
                }
            }
            for (int v = 0; v < fx[i].length; v++) {
                if (fx[i][v] == -1) {
                    fx[i][v] = max;
                }
            }
        }
        for (int i = 0; i < fx[0].length; i++) {
            for (int j = 0; j < fx.length; j++) {
                rx[i][j] = fx[j][i];
            }
        }
        return rx;
    }
}

File: PYTHON/EASY_PY_LEETCODE/PROB_3033_Modify_The_Matrix.py
from typing import *
class Solution:
    def modifiedMatrix(self, matrix: List[List[int]]) -> List[List[int]]:
        fx , rx = [] , []
        for j in range(len(matrix[0])):
            b = []
            for k in range(len(matrix)):
                b.append(matrix[k][j])
            fx.append(b)
        for l in fx:
            for v in range(len(l)):
                if l[v] == -1:
                    l[v] = max(l)
        for m in range(len(fx[0])):
            r = []
            for n in range(len(fx)):
                r.append(fx[n][m])
            rx.append(r)
        return rx